plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java

```diff
@@ -330,20 +331,37 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return new ConnectorTableProperties();
     }
 
+    @Override
+    public ConnectorTableBasicMetadata getTableBasicMetadata(ConnectorSession session, ConnectorTableHandle table)
+    {
+        JdbcTableHandle handle = (JdbcTableHandle) table;
+
+        return new ConnectorTableBasicMetadata(
+                getSchemaTableName(handle),
+                jdbcClient.getColumns(session, handle).stream()
+                        .map(JdbcColumnHandle::getColumnBasicMetadata)
+                        .collect(toImmutableList()));
+    }
+
     @Override
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
     {
         JdbcTableHandle handle = (JdbcTableHandle) table;
 
-        ImmutableList.Builder<ColumnMetadata> columnMetadata = ImmutableList.builder();
-        for (JdbcColumnHandle column : jdbcClient.getColumns(session, handle)) {
-            columnMetadata.add(column.getColumnMetadata());
-        }
-        SchemaTableName schemaTableName = handle.isNamedRelation()
+        return new ConnectorTableMetadata(
+                getSchemaTableName(handle),
+                jdbcClient.getColumns(session, handle).stream()
+                        .map(JdbcColumnHandle::getColumnMetadata)
+                        .collect(toImmutableList()),
+                jdbcClient.getTableProperties(session, handle));
+    }
+
+    protected static SchemaTableName getSchemaTableName(JdbcTableHandle handle)
```

Comment:
This method is extracted from a hack (see TODO inside) and hopefully can stay `private`, unless you already see a need to expose this.

Reply:
> unless you already see a need to expose this.

I do, phoenix connector potentially can also have `!handle.isNamedRelation()`, but it was missing a TODO like this there.

Reply:
Maybe as a separate commit ?